pkg/plugin/loader/loader.go

```diff
@@ -53,98 +51,142 @@ func LoadAllPlugins(namespace string, searchPath []string, selections []plugin.S
 			continue
 		}
 
-		definitions, err := scanPlugins(dir)
+		files, err := findPlugins(dir)
 		if err != nil {
-			return ret, err
+			return []plugin.Interface{}, errors.Wrapf(err, "couldn't scan %v for plugins", dir)
 		}
+		pluginFiles = append(pluginFiles, files...)
+	}
 
-		defns = append(defns, definitions...)
+	pluginDefs := []*pluginDefinition{}
+	for _, file := range pluginFiles {
+		pluginDef, err := loadDefinition(file)
+		if err != nil {
+			return []plugin.Interface{}, errors.Wrapf(err, "couldn't load plugin definition %v", file)
+		}
+		pluginDefs = append(pluginDefs, pluginDef)
 	}
 
-	for _, selection := range selections {
-		for _, pluginDef := range defns {
-			if selection.Name == pluginDef.Name {
-				p, err := loadPlugin(namespace, pluginDef, masterAddress)
-				if err != nil {
-					return ret, err
-				}
-				ret = append(ret, p)
-			}
+	pluginDefs = filterPluginDef(pluginDefs, selections)
+
+	plugins := []plugin.Interface{}
+	for _, def := range pluginDefs {
+		pluginIface, err := loadPlugin(def, namespace)
+		if err != nil {
+			return nil, errors.Wrapf(err, "couldn't load plugin %v", def.SonobuoyConfig.PluginName)
 		}
+		plugins = append(plugins, pluginIface)
 	}
-	return ret, nil
+
+	return plugins, nil
 }
 
 // loadPlugin loads an individual plugin by instantiating a plugin driver with
 // the settings from the given plugin definition and selection
-func loadPlugin(namespace string, dfn plugin.Definition, masterAddress string) (plugin.Interface, error) {
-	// TODO(chuckha): We don't use the cfg for anything except passing a string around. Consider removing this struct.
-	cfg := &plugin.WorkerConfig{}
-	logrus.Infof("Loading plugin driver %v", dfn.Driver)
-	switch dfn.Driver {
-	case "DaemonSet":
-		cfg.MasterURL = "http://" + masterAddress + "/api/v1/results/by-node"
-		return daemonset.NewPlugin(namespace, dfn, cfg), nil
-	case "Job":
-		cfg.MasterURL = "http://" + masterAddress + "/api/v1/results/global"
-		return job.NewPlugin(namespace, dfn, cfg), nil
-	default:
-		return nil, errors.Errorf("Unknown driver %v", dfn.Driver)
-	}
+// func loadPlugin(namespace string, dfn plugin.Definition, masterAddress string) (plugin.Interface, error) {
+// 	// TODO(chuckha): We don't use the cfg for anything except passing a string around. Consider removing this struct.
+// 	cfg := &plugin.WorkerConfig{}
+// 	logrus.Infof("Loading plugin driver %v", dfn.Driver)
+// 	switch dfn.Driver {
+// 	case "DaemonSet":
+// 		cfg.MasterURL = "http://" + masterAddress + "/api/v1/results/by-node"
+// 		return daemonset.NewPlugin(namespace, dfn, cfg), nil
+// 	case "Job":
+// 		cfg.MasterURL = "http://" + masterAddress + "/api/v1/results/global"
+// 		return job.NewPlugin(namespace, dfn, cfg), nil
+// 	default:
+// 		return nil, errors.Errorf("Unknown driver %v", dfn.Driver)
+// 	}
+// }
+
+func findPlugins(dir string) ([]string, error) {
+	return filepath.Glob(path.Join(dir, "*.yml"))
```

Comment:
nit: I think kubernetes folks tend to use `*.yaml` -- but if there is a reason we need a 3 letter file extension, please comment.

Reply:
let's do both!

Reply:
![both](https://user-images.githubusercontent.com/98927/35525080-78704bdc-04f1-11e8-9ac5-6c1dd61c37d7.gif)